src/Documentation/Frequently-Asked-Questions.md

```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
+Those entities are generally isolated from each other, can be accessed and updated independently, and expose a well defined set of supported operations.
+This works well with the intuitive "one entity - one grain" modelling.
+
+An application entity may be too big to be efficiently represented by a single grain if it encapsulates too much state, and as a result has to handle a high rate of requests to it.
+Even a single grain can generally handle up to a few thousand of trivial calls per second, the rule of thumb is to be wary of individual grain receiving hundreds of requests per second.
+That may be a sign of the grain being too large, and decomposing it into a set of smaller grains may lead to a more stable and balanced system.
+
+An application entity may be too small to be a grain if that would cause constant interaction of other grains with it, and as a result, to too much of messaging overhead.
+In such cases, it may make more sense to make those closely interacting entities part of a single grain, so that they would invoke each other directly.
+
+### How should you avoid grain hot spots?
 
 The throughput of a grain is limited by a single thread that its activation can execute on.
-Therefore, it is advisable to avoid designs where a single grain receives a disproportionate share of requests.
+Therefore, it is advisable to avoid designs where a single grain receives a disproportionate share of requests or is involved in processing requests to other grains.
 There are various patterns that help prevent overloading of a single grain even when logically it is a central point of communication.
 
 For example, if a grain is an aggregator of some counters or statistics that are reported by a large number of grains on a regular basis, one proven approach is to add a controlled number of intermediate aggregator grains and assign each of the reporting grains (using a modulo on a key or a hash) to an intermediate aggregator, so that the load is more or less evenly distributed across all intermediate aggregator grains that in their turn periodically report partial aggregates to the central aggregator grain.
 
-## How do I tear down a grain?
+### Can a single Orleans cluster run across multiple datacenters?
+
+Orleans clusters are currently limited to a single data center per cluster.
+Instead, since 1.3.0, you can consider a [multi-cluster deployment where clusters deployed to different datacenters form a single multi-cluster](Multi-Cluster/Overview.md). 
+
+### In what cases can a split brain (same grain activated in multiple silos at the same time) happen?
+
+During normal operations the Orleans runtime guarantees that each grain will have at most one instance in the cluster.
+The only time this guarantee can be violated is when a silo crashes or gets killed without a proper shutdown.
+In that case, there is a ~30 second (based on configuration) window where a grain can potentially get temporarily instantiated in more than one silo.
```

Comment:
Maybe add "Convergence to a single activation per grain is guaranteed and duplicate activations will be deactivated when this window closes."